plugins/communications-app/TaskAlertModalForm/index.jsx

```diff
@@ -0,0 +1,160 @@
+
+  const handlePostEmailTask = async () => {
+    const emailData = new FormData();
+    emailData.append('action', 'send');
+    emailData.append('send_to', JSON.stringify(emailRecipients));
+    emailData.append('subject', subject);
+    emailData.append('message', body);
+
+    if (isScheduled) {
+      emailData.append('schedule', new Date(`${scheduleDate} ${scheduleTime}`).toISOString());
+    }
+
+    changeFormStatus('pending');
+
+    try {
+      await postBulkEmailInstructorTask(emailData, courseId);
+      const newFormStatus = isScheduled ? 'completeSchedule' : 'complete';
+      changeFormStatus(newFormStatus);
+      setTimeout(() => changeFormStatus('default'), 3000);
+    } catch {
+      changeFormStatus('error');
+    }
+  };
+
+  const handlePatchEmailTask = async () => {
+    const emailRecipientsValue = emailRecipients;
+    const emailSubject = subject;
+    const emailBody = body;
+
+    const emailData = {
+      email: {
+        targets: emailRecipientsValue,
+        subject: emailSubject,
+        message: emailBody,
+        id: emailId,
+      },
+      schedule: isScheduled ? new Date(`${scheduleDate} ${scheduleTime}`).toISOString() : null,
+    };
+
+    changeFormStatus('pending');
+
+    try {
+      await patchScheduledBulkEmailInstructorTask(emailData, courseId, schedulingId);
+      changeFormStatus('completeSchedule');
+      setTimeout(() => changeFormStatus('default'), 3000);
+    } catch {
+      changeFormStatus('error');
+    }
+  };
+
+  const createEmailTask = async () => {
+    const isScheduleValid = isScheduled ? scheduleDate.length > 0 && scheduleTime.length > 0 : true;
+    const isFormValid = emailRecipients.length > 0 && subject.length > 0
+    && body.length > 0 && isScheduleValid;
+
+    if (isFormValid && isEditMode) {
+      await handlePatchEmailTask();
+    }
+
+    if (isFormValid && !isEditMode) {
+      await handlePostEmailTask();
+    }
+
+    if (isFormValid) {
+      dispatch(getScheduledBulkEmailThunk(courseId, 1));
+    }
+  };
+
+  return (
+    <TaskAlertModal
+      isOpen={isTaskAlertOpen}
+      alertMessage={isEditMode
+        ? (
+          <EditMessage
+            subject={subject}
+            emailRecipients={emailRecipients}
+            {...{ scheduleDate, scheduleTime, isScheduled }}
+          />
+        )
+        : (
+          <AlertMessage
+            subject={subject}
+            emailRecipients={emailRecipients}
+            isScheduled={isScheduled}
+          />
+        )}
+      close={(event) => {
+        closeTaskAlert();
+
+        if (event.target.name === 'continue') {
+          if (!isFormSubmitted) {
+            dispatchForm(formActions.updateForm({ isFormSubmitted: true }));
+          }
+
+          if (isScheduleButtonClicked) {
+            dispatchForm(formActions.updateForm({ isScheduledSubmitted: true }));
+          }
+
+          createEmailTask();
+        }
+      }}
+    />
```

Comment:
I would suggest to move this into another fuction


```suggestion
      close={handleClose}
    />
```